data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeEvaluator.java

```diff
@@ -37,7 +37,7 @@ public Object evaluate(ParseTree parseTree, Event event) {
             walker.walk(listener, parseTree);
             return listener.getResult();
         } catch (final Exception e) {
-            LOG.error("Unable to evaluate event", e);
+            LOG.error(e.getMessage());
```

Comment:
We should log the stack trace when catching `Exception`.

Please add new catch statements for known exceptions (e.g. parse error) and log those without the stack trace.